C2CRIBuildDir/projects/C2C-RI/src/RI_Logging/src/org/fhwa/c2cri/logger/RILogging.java

```diff
@@ -140,31 +148,31 @@ public void configureLogging(String logFileName, String configFileName, String l
             DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
             Date date = new Date();
 
-            logFile = logFileName + "." + dateFormat.format(date) + ".xml";
-
-            log = Logger.getLogger("net.sf.jameleon");
+	LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+	logFile = logFileName + "." + dateFormat.format(date) + ".xml";
+        RIXMLLayout oLayout = RIXMLLayout.createLayout();
+        oLayout.m_bUseCdata = false;
+	riAppender = FileAppender.newBuilder().withFileName(logFile).setName("riAppender").setImmediateFlush(true).setBufferedIo(true).setLayout(oLayout).build();
+	riAppender.start();
+        Logger oRoot = (Logger)LogManager.getRootLogger();
+        oRoot.addAppender(riAppender);
+        oRoot.addAppender(riGUIAppender);
+			
+
+            log = (Logger)LogManager.getLogger("C2CRIDebug");
             log.addAppender(riAppender);
             log.addAppender(riGUIAppender);
-            log2 = Logger.getLogger("org.fhwa.c2cri");
-            log2.addAppender(riAppender);
-            log2.addAppender(riGUIAppender);
-            riAppender.setName("STDOUT");
-            riAppender.setFile(logFile);
-            riAppender.setThreshold(Level.INFO);
-            RIXMLLayout xmlLayout = new RIXMLLayout();
-            riAppender.setLayout(xmlLayout);
-            riAppender.setBufferedIO(true);
-            riAppender.activateOptions();
-            riGUIAppender.setThreshold(Level.INFO);
```

Comment:
Do we need to set threshold for the new appenders

Reply:
Thresholds are set on the Logger, not the Appender, in Log4j v2